src/main/scala/com/raphtory/core/deploy/Raphtory.scala

```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
 import scala.reflect.ClassTag
 import scala.reflect.runtime.universe._
 
+/**
+  * {s}`Raphtory`
+  *  : `Raphtory` object for creating Raphtory Components
+  *
+  * ## Methods
+  *
+  *   {s}`createGraph(spout: Spout[T] = new IdentitySpout[T](), graphBuilder: GraphBuilder[T], customConfig: Map[String, Any] = Map()): RaphtoryGraph[T]`
+  *    : Create Graph using spout, graph-builder and custom config
+  *
+  *   {s}`createClient(deploymentID: String = "", customConfig: Map[String, Any] = Map())`
+  *    : Create Client
+  *
+  *   {s}`createSpout(spout: Spout[T])`
+  *    : Create Spout
+  *
+  *   {s}`createGraphBuilder(builder: GraphBuilder[T])`
+  *    : Create Graph Builder
+  *
+  *   {s}`createPartitionManager()`
+  *    : Creates Partition Manager
+  *
+  *   {s}`createQueryManager()`
+  *    : Creates Query Manager
+  *
+  *   {s}`getDefaultConfig(customConfig: Map[String, Any] = Map()): Config`
+  *    : Returns default config
+  *
+  *   {s}`confBuilder(customConfig: Map[String, Any] = Map()): Config`
+  *    : Returns Config
```

Comment:
the conf builder is pretty important, its used almost everywhere, can we please explain where is the config this returns? how is that generated? what kind of custom config can we use? can we have an example of what a custom config is?